### PR TITLE
Issue945 ppd bypass

### LINKED ISSUE #945: PPD12 bypass
The ventilation unit of PPD12 should have a bypass damper.

### PR DESCRIPTION
**Pull request by Mathadon**
this closes #945

### CODE REVIEW

**Comment by Mathadon** [comment]
#943 should be merged first

**Comment by Mathadon** [comment]
@damienpicard this is ready to be merged

File: IDEAS/Examples/PPD12/Structure.mo [hunk 1]
```diff
@@ -225,7 +229,9 @@ model Structure "Ppd 12 example model"
     bouTypA=IDEAS.Buildings.Components.Interfaces.BoundaryType.External,
     nSurfExt=2,
     redeclare Data.Ppd12WestShadingFirst shaTypC,
-    n50=n50) "Master bedroom"
+    n50=n50,
+    mSenFac=1)
+             "Master bedroom"
```
Comment (damienpicard) [inline IDEAS/Examples/PPD12/Structure.mo]: lol

**Comment by damienpicard** [comment]
Where is the kitchen?

**Comment by damienpicard** [comment]
I'm not sure how you designed your pressure drop in the hydraulic system but keep in mind that fittings and bends are responsible for a large part of the pressure drop.

**Comment by damienpicard** [comment]
Once my minor comments are answered, this can be merged.

**Comment by damienpicard** [comment]
It might be nice to include the validation data into IDEAS as well?

File: IDEAS/Examples/PPD12/BaseClasses/VentilationNoControl.mo [hunk 2]
```diff
@@ -18,28 +18,30 @@ partial model VentilationNoControl "Ppd 12 example model"
...
     energyDynamics=Modelica.Fluid.Types.Dynamics.SteadyState,
     m_flow_nominal=m_flow_nominal_air,
-    inputType=IDEAS.Fluid.Types.InputType.Constant,
     tau=0,
     use_inputFilter=false,
     dp_nominal=300,
     redeclare package Medium = MediumAir,
-    constantMassFlowRate=70*1.2/3600)     "Return fan"
-    annotation (Placement(transformation(extent={{320,170},{340,190}})));
+    constantMassFlowRate=70*1.2/3600,
```
[the remark this answers is not in the document]
Comment (Mathadon) [reply IDEAS/Examples/PPD12/BaseClasses/VentilationNoControl.mo]: I think it's fine as it is?

File: IDEAS/Examples/PPD12/Heating.mo [hunk 3]
```diff
@@ -3,18 +3,22 @@ model Heating "Ppd 12 example model"
...
     "Conversion block of control signal to pump pressure set point"
     annotation (Placement(transformation(extent={{280,-80},{300,-60}})));
-  Modelica.Blocks.Continuous.Integrator integrator(k=1/3600000)
-    annotation (Placement(transformation(extent={{350,-66},{370,-46}})));
+  Modelica.Blocks.Continuous.Integrator EHea(k=1/3600000) "Heat energy meter"
+    annotation (Placement(transformation(extent={{380,-62},{400,-42}})));
   Modelica.Blocks.Sources.Constant Thea(k=273.15 + 70)
     "Supply water temperature set point"
     annotation (Placement(transformation(extent={{400,-104},{380,-84}})));
+  Modelica.Blocks.Continuous.Integrator EGas(k=1/3600000) "Gas energy meter"
```
[the remark this answers is not in the document]
Comment (Mathadon) [reply IDEAS/Examples/PPD12/Heating.mo]: for the output `y` or?

File: IDEAS/Examples/PPD12/Structure.mo (hunk 1 above)
Comment (Mathadon) [reply IDEAS/Examples/PPD12/Structure.mo]: that's how it's called.. 🙄

File: IDEAS/Examples/PPD12/BaseClasses/HeatingNoControl.mo [hunk 4]
```diff
@@ -116,61 +131,61 @@ partial model HeatingNoControl "Ppd 12 example model without control"
     massDynamics=Modelica.Fluid.Types.Dynamics.SteadyState,
     redeclare package Medium = MediumWater,
     m_flow_nominal={m_flow_nominal,m_flow_nominal,m_flow_nominal},
-    dp_nominal=2*{0,0,dp_16mm*5},
     portFlowDirection_1=Modelica.Fluid.Types.PortFlowDirection.Entering,
     portFlowDirection_2=Modelica.Fluid.Types.PortFlowDirection.Leaving,
-    portFlowDirection_3=Modelica.Fluid.Types.PortFlowDirection.Leaving)
+    portFlowDirection_3=Modelica.Fluid.Types.PortFlowDirection.Leaving,
+    dp_nominal=2*{0,0,0})
```
[the remark this answers is not in the document]
Comment (Mathadon) [reply IDEAS/Examples/PPD12/BaseClasses/HeatingNoControl.mo]: > Where is the kitchen?

in the `Diner` zone

File: IDEAS/Examples/PPD12/Heating.mo (hunk 3 above)
[the remark this answers is not in the document]
Comment (damienpicard) [reply IDEAS/Examples/PPD12/Heating.mo]: For EGas. Can be in the comment

File: IDEAS/Examples/PPD12/BaseClasses/VentilationNoControl.mo (hunk 2 above)
[the remark this answers is not in the document]
Comment (damienpicard) [reply IDEAS/Examples/PPD12/BaseClasses/VentilationNoControl.mo]: yes, except that this is a conversion factor we use on and on. So why not adding it in the constants? But I'm fine with the current implementation.

**Comment by Mathadon** [comment]
fixed through #957

**Pull request closed without merging**